watertap/tools/parameter_sweep.py

```diff
@@ -444,32 +688,74 @@ def parameter_sweep(model, sweep_params, outputs, results_file=None, optimize_fu
             local_results[k, :] = [pyo.value(outcome) for outcome in outputs.values()]
             previous_run_failed = False
 
-        if previous_run_failed and (reinitialize_function is not None):
-            # We choose to re-initialize the model at this point
+        # If the initial attempt failed and additional conditions are met, try
+        # to reinitialize and resolve.
+        if reinitialize_before_sweep == False and previous_run_failed and (reinitialize_function is not None):
             try:
                 reinitialize_function(model, **reinitialize_kwargs)
-                optimize_function(model, **optimize_kwargs)
+                results = optimize_function(model, **optimize_kwargs)
+                pyo.assert_optimal_termination(results)
+
             except:
-                # do we raise an error here?
-                # nothing to do
+                # solver_termination_condition = "other" # Delete Me
+                # solver_termination_condition = results.solver.termination_condition.name
                 pass
             else:
                 local_results[k, :] = [pyo.value(outcome) for outcome in outputs.values()]
+                # solver_termination_condition = results.solver.termination_condition.name
 
 
-    # ================================================================
-    # Save results
-    # ================================================================
+        elif reinitialize_before_sweep and previous_run_failed:
+            pass
+
+        # Update the loop based on the reinitialization
+        solver_termination_condition = results.solver.termination_condition.name
+        _update_local_output_dict(model, sweep_params, k, local_values[k, :], solver_termination_condition, local_output_dict)
```

Comment:
```suggestion
        solver_termination_condition = results.solver.termination_condition.name
        _update_local_output_dict(model, sweep_params, k, local_values[k, :], solver_termination_condition, local_output_dict)
```

If you take some of the suggestions above.

Reply:
incorporated
https://github.com/watertap-org/watertap/blob/666d10ab5414fd603a4003b9e34402b35acfa372/watertap/tools/parameter_sweep.py#L681